lib/OperationHandler.hpp

```diff
@@ -53,13 +54,13 @@ class OperationHandler: private INotificationHandler
     DisconnectResponse HandleDisconnectRequest(const DisconnectRequest& disconnectRequest);
     ScanResponse HandleScanRequest(const ScanRequest& scanRequest);
 
-    using GuestNotificationCallback = std::function<void(std::variant<DisconnectNotif, SignalQualityNotif>)>;
```

Comment:
I like keeping types relatively explicit when possible, as long as they are not unreasonably long.
Knowing this is an std::function and the return type / parameter types directly is convenient, instead of having to lookup what a `GuestNotificationCallback` is, then what a `GuestNotificationTypes` is.